src/app/im-handler.cpp

```diff
@@ -0,0 +1,477 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
```

Comment:
Should add a clear comment saying that the file is generated with ZAP to prevent confusion. Comment should be added in all of the templates. Best way to do so would be to add the comment directly in the `header.zapt` This will clear out part of the confusion caused by having generated files in the `src/app/` directory